test/test_framework.py

```diff
@@ -612,6 +612,104 @@ class CustomObject:
 
             framework_copy.commit()
 
+    def test_set_operations(self):
+        class SomeObject(Object):
+            state = StoredState()
+
+            def __init__(self, framework, key):
+                super().__init__(framework, key)
+
+        test_operations = [(
+            {"1"},
+            lambda: obj.state.set | other_set,
+            lambda: (
+                self.assertEqual(res_set, {"1", "a", "b"}),
+                self.assertEqual(obj.state.set._under, {"a", "b"}),
+                self.assertEqual(other_set, set(["1"])),
```

Comment:
There still seems to be a lot of repetition in these checks.  Can I suggest something like this instead:

```python
        tests = [
            # input           operation            (a, b) result    (b, a) result
            ({"1"},           lambda a, b: a | b,  {"1", "a", "b"}, {"1", "a", "b"}),
            ({"a", "c"},      lambda a, b: a - b,  {"c"},           {"b"}),
            ({"a", "c"},      lambda a, b: a & b,  {"a"},           {"a"}),
            ({"a", "c", "d"}, lambda a, b: a ^ b,  {"b", "c", "d"}, {"b", "c", "d"}),
            (set(),           lambda a, b: set(a), {"a", "b"},      set()),
        ]

        framework = self.create_framework()

        # Validate that operations between StoredSet and built-in sets only result in built-in sets being returned.
        # Make sure that commutativity is preserved and that the original sets are not changed or used as a result.
        for input, operation, ab_res, ba_res in test_operations:
            obj = SomeObject(framework, "2")
            obj.state.set = {"a", "b"}

            for a, b, expected in [(obj.state.set, input, ab_res), (input, obj.state.set, ba_res)]:
                old_a = set(a)
                old_b = set(b)

                result = operation(a, b)
                self.assertEqual(result, expected)

                # Common sanity checks
                self.assertIsNot(obj.state.set._under, result)
                self.assertIsNot(res, a)
                self.assertIsNot(res, b)
                self.assertEqual(a, old_a)
                self.assertEqual(b, old_b)
```